central/compliance/datastore/internal/store/postgres/store.go

```diff
@@ -0,0 +1,249 @@
+	if err != nil {
+		return nil, err
+	}
+	results, exists, err := s.results.Get(ctx, metadata.GetRunId())
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errox.NotFound.Newf("run results with id %q was not found", metadata.GetRunId())
+	}
+
+	results.RunMetadata = metadata
+	results.Domain = domain
+
+	if flags&(dsTypes.WithMessageStrings|dsTypes.RequireMessageStrings) != 0 {
+		externalizedStrings, exists, err := s.strings.Get(ctx, metadata.GetRunId())
+		if err != nil || !exists {
+			if flags&dsTypes.RequireMessageStrings != 0 {
+				return nil, errors.Wrap(err, "loading message strings")
+			}
+			log.Errorf("Could not load message strings for compliance run results: %v", err)
+		}
+		if !store.ReconstituteStrings(results, externalizedStrings) {
+			return nil, errors.New("some message strings could not be loaded")
+		}
+	}
+	return results, nil
+}
+
+func (s *storeImpl) GetSpecificRunResults(ctx context.Context, _, _, runID string, flags types.GetFlags) (types.ResultsWithStatus, error) {
+	metadata, exists, err := s.metadata.Get(ctx, runID)
+	if err != nil {
+		return types.ResultsWithStatus{}, err
+	}
+	if !exists {
+		return types.ResultsWithStatus{}, errox.NotFound.Newf("run metadata with id %q was not found", runID)
+	}
+	if !metadata.GetSuccess() {
+		return types.ResultsWithStatus{
+			FailedRuns: []*storage.ComplianceRunMetadata{
+				metadata,
+			},
+		}, nil
+	}
+	results, err := s.getResultsFromMetadata(ctx, metadata, flags)
+	if err != nil {
+		return types.ResultsWithStatus{}, err
+	}
+	return types.ResultsWithStatus{
+		LastSuccessfulResults: results,
+	}, nil
+}
+
+func (s *storeImpl) GetLatestRunResults(ctx context.Context, clusterID, standardID string, flags types.GetFlags) (types.ResultsWithStatus, error) {
+	metadataBatch, err := s.getLatestRunMetadata(ctx, clusterID, standardID)
+	if err != nil {
+		return types.ResultsWithStatus{}, err
+	}
+	if metadataBatch.LastSuccessfulRunMetadata == nil {
+		return types.ResultsWithStatus{
+			FailedRuns: metadataBatch.FailedRunsMetadata,
+		}, nil
+	}
+	results, err := s.getResultsFromMetadata(ctx, metadataBatch.LastSuccessfulRunMetadata, flags)
+	if err != nil {
+		return types.ResultsWithStatus{}, err
+	}
+	return types.ResultsWithStatus{
+		LastSuccessfulResults: results,
+		FailedRuns:            metadataBatch.FailedRunsMetadata,
+	}, nil
+}
+
+func (s *storeImpl) GetLatestRunResultsBatch(ctx context.Context, clusterIDs, standardIDs []string, flags types.GetFlags) (map[compliance.ClusterStandardPair]types.ResultsWithStatus, error) {
+	pairsToResults := make(map[compliance.ClusterStandardPair]types.ResultsWithStatus)
+	for _, clusterID := range clusterIDs {
+		for _, standardID := range standardIDs {
+			results, err := s.GetLatestRunResults(ctx, clusterID, standardID, flags)
+			if err != nil {
+				return nil, err
+			}
+			pairsToResults[compliance.ClusterStandardPair{ClusterID: clusterID, StandardID: standardID}] = results
```

Comment:
This is either a nit OR an opportunity for me to learn something.  Why do you set the map like this here but down below use the NewPair down below?
`results[compliance.NewPair(clusterID, standardID)] = metadata`

Reply:
because I messed up :)